integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,254 @@
+        total_projects = folderless_projects + foldered_projects
+
+        if not total_projects:
+            logger.warning("Space query returned 0 projects")
+
+        params["page_size"] = PAGE_SIZE
+        page = 0
+        while page * PAGE_SIZE < len(total_projects):
+            logger.info(f"Current query position: page {page}")
+
+            folder_response_list = (await self._get_folders(space_id, params))["folders"]
+            folderless_projects = (await self._get_folderless_projects(space_id, params))["lists"]
+            foldered_projects = ClickUpClient.parse_projects_from_folders(folder_response_list)
+
+            projects = folderless_projects + foldered_projects
+            yield projects
+            page += 1
+
+    async def get_paginated_teams(self) -> AsyncGenerator[List[Dict[str, Any]], None]:
+        """
+        Fetches and yields a list of teams from the ClickUp API.
+
+        Yields:
+            list[dict[str, Any]]: Lists of team dictionaries.
+        """
+        logger.info("Getting teams from ClickUp")
+        total_teams = (await self._get_teams())["teams"]
+
+        if not total_teams:
+            logger.warning(
+                "Team query returned 0 teams, did you provide the correct ClickUp API credentials?"
+            )
+
+        yield total_teams
+
+    async def get_paginated_spaces(
+        self, team_id: str
+    ) -> AsyncGenerator[List[Dict[str, Any]], None]:
+        """
+        Fetches and yields paginated lists of spaces for a given team.
+
+        Args:
+            team_id (str): The ID of the team.
+
+        Yields:
+            list[dict[str, Any]]: Lists of space dictionaries.
+        """
+        logger.info("Getting spaces from ClickUp")
+
+        params = self._generate_base_req_params()
+        total_spaces = (await self._get_spaces(team_id, params))["spaces"]
+
+        if not total_spaces:
+            logger.warning(
+                "Space query returned 0 spaces, did you provide the correct ClickUp API credentials?"
+            )
+
+        params["page_size"] = PAGE_SIZE
+        page = 0
+        while page * PAGE_SIZE < len(total_spaces):
+            logger.info(f"Current query position: page {page}")
+            space_response_list = (await self._get_spaces(team_id, params))["spaces"]
+            yield space_response_list
+            page += 1
+
+    async def get_paginated_tasks(
+        self, list_id: str
+    ) -> AsyncGenerator[List[Dict[str, Any]], None]:
+        """
+        Fetches and yields paginated lists of tasks for a given list.
+
+        Args:
+            list_id (str): The ID of the list.
+
+        Yields:
+            list[dict[str, Any]]: Lists of task dictionaries.
+        """
+        logger.info("Getting tasks (issues) from ClickUp")
+
+        params = self._generate_base_req_params()
+        total_tasks = (await self._get_tasks(list_id, params))["tasks"]
```

Comment:
When you get_tasks, Clickup returns the first 100 records. So if you are paginating based on total_task, this will not yield the expected result. Try to have about 120 issues in Clickup and see how this works - to see if you can display all the 120 issues to port